src/setup/run.go

```diff
@@ -161,9 +159,24 @@ func ProvisionFunctionsServerlessAzure(config *Configuration, serverlessDirPath
 			config.SubExperiments[subExperimentIndex].Endpoints = []EndpointInfo{}
 		}
 
-		for parallelism := 0; parallelism < subExperiment.Parallelism; parallelism++ {
+		deploySubExperimentParallelismInBatches(config, serverlessDirPath, randomExperimentTag, subExperimentIndex)
+	}
+}
+
+func deploySubExperimentParallelismInBatches(config *Configuration, serverlessDirPath string, randomExperimentTag string, subExperimentIndex int) {
+	subExperiment := config.SubExperiments[subExperimentIndex]
+
+	const functionsPerBatch = 3
```

Comment:
Could be useful to make `functionsPerBatch` a function parameter instead to make it clear that we can change this